### PR TITLE
Modernized ES modules in RecoTracker/MeasurementDet

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

- Added fillDescriptions
   - This required removing unused parameters from config files
- Used ESGetToken and consumes were needed
- Have ParameterSet only used by ESProducer and not helpers
- Combined header and source for ESProducer and removed modules.cc

#### PR validation:

Ran workflow 4.17 successfully after the change.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc [hunk 1]
```diff
@@ -57,111 +167,84 @@ std::unique_ptr<MeasurementTracker> MeasurementTrackerESProducer::produce(const
...
+
+    desc.addOptional<edm::ParameterSetDescription>("badStripCuts", mtiDesc);
   }
+
+  desc.add<bool>("UsePixelModuleQualityDB", true)->setComment("Use DB info at the module level (that is, detid level)");
+  desc.addUntracked<bool>("DebugPixelModuleQualityDB", false)->setComment("dump out info on module status");
+  desc.add<bool>("UsePixelROCQualityDB", true)->setComment("Use DB info at the ROC level");
+  desc.addUntracked<bool>("DebugPixelROCQualityDB", false)->setComment("dump out info om module status ");
+
+  descriptions.addDefault(desc);
```
Comment (slava77) [inline RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: what is the reason there is no named description to replace MeasurementTrackerESProducer_cfi ?

Comment (Dr15Jones) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: As the Core is adding `consumes` to all ESProducers, we've tried to minimize the possible impact of our changes (although sometimes we just can't help ourselves and feel compelled to re-write more code then necessary :) ). So I didn't want to touch the cfi.py files actually loaded in production. Plus MeasurementTrackerESProducer_cfi.py uses an `Era` which can not be done by the `fillDescriptions`.

Comment (slava77) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: I worry that this is opening up a problem of large and complex duplication of the configuration with different implementation, which is bound to quickly diverge and become confusing.

Comment (Dr15Jones) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: I can remove all the default values if you prefer.

Comment (perrotta) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: ... and confuse HLT parsing as well.

One could replace courrent _cfi with a _cff one, which imports the auto-generated one and can still implemet the Era's. If so, one should also propagate whenever there is an inclusion of MeasurementTrackerESProducer_cfi.py  with the new MeasurementTrackerESProducer_cff.py

Comment (perrotta) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: Or even let the fillDescriptions define a MeasurementTrackerESProducerDefault_cfi (or similar)

Comment (Dr15Jones) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: I have been considering having the `addDefault` actually allow generation of a `classtype_default.py` file which would declare a variable with name `_classtype_default`. The `_` guarantees that an `import *` statement or doing `process.load` would not pick it up accidentally but it still could be used for cloning.

Comment (slava77) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: thinking of some other ways we were getting to the era-modified cases, this pattern seems reasonable to me (a generated cfi has a clearly distinct, not intuitively usable general name, the manually created cfi or cff would load it and clone to the usable name and also have the modifiers applied as necessary)

**Comment by slava77** [comment]
@Dr15Jones 
please clarify if you are planning to update this PR (in line with https://github.com/cms-sw/cmssw/pull/27589#discussion_r307349832 ).
Thank you.

**Comment by perrotta** [comment]
(Similar issues for #27594 and #27596)

**Comment by Dr15Jones** [comment]
Sorry, I was waiting for @slava77 and @perrotta to decide how you wanted to handle the fillDescriptions as it is code you'd have to maintain. From my side, doing `addDefault` is the least intrusive way to improve the code.

File: RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc (hunk 1 above)
Comment (Dr15Jones) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: So what do you want me to do?
1. leave as is
1. remove all default values in fillDescriptions except "Phase2StripCPE" which was previously using the `existsAs` trick to create a default value for a tracked parameter (which we do NOT want people to do)
1. have the fillDescriptions write a `MeasurementTrackerESProducerDefault_cfi.py` file and then change `MeasurementTrackerESProducer_cfi.py` to load that file in such a way as to not create 2 modules

Comment (slava77) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: I do not want to have \~30 parameters (with somewhat complicated nested structure) defined in two places with some expectations of identity.
[the limit where it still seems acceptable to me is around O(1) parameter]

The proposal to generate ```classtype_default.py``` (perhaps still a \*cfi.py) looks good to me (perhaps I was not clear in https://github.com/cms-sw/cmssw/pull/27589#discussion_r307390394).
Please proceed.
Thank you.

Comment (Dr15Jones) [reply RecoTracker/MeasurementDet/plugins/MeasurementTrackerESProducer.cc]: What I had proposed for `classtype_default.py` was a possible future extension to the parameter set system and would not be part of these pull requests. The 'option 3' (make a `MeasurementTrackerESProducerDefault_cfi.py` file using fillDescriptions) approximates that behavior based on what is possible now.

I will do 'option 3'.

**Comment by Dr15Jones** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/27589	e65424c
- code changes are in line with the PR description and the follow up review
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

MeasurementTracker in  RecoTracker/MeasurementDet/python/MeasurementTrackerESProducer_cfi.py is now appropriately derived from the fillDescriptions

**Comment by kpedro88** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**